dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layerparam/initializer/ConvolutionalLayerParameterInitializer.java

```diff
@@ -0,0 +1,125 @@
+  private final int paddingHeight;
+  private final int paddingWidth;
+  private final int strideHeight;
+  private final int strideWidth;
+  private final int kernelHeight;
+  private final int kernelWidth;
+  private final float initWeight;
+  private final float initBias;
+  private final long randomSeed;
+
+  @Inject
+  private ConvolutionalLayerParameterInitializer(
+      final MatrixFactory matrixFactory,
+      @Parameter(LayerConfigurationParameters.LayerIndex.class) final int index,
+      @Parameter(LayerConfigurationParameters.LayerInputShape.class) final String inputShape,
+      @Parameter(LayerConfigurationParameters.PaddingHeight.class) final int paddingHeight,
+      @Parameter(LayerConfigurationParameters.PaddingWidth.class) final int paddingWidth,
+      @Parameter(LayerConfigurationParameters.StrideHeight.class) final int strideHeight,
+      @Parameter(LayerConfigurationParameters.StrideWidth.class) final int strideWidth,
+      @Parameter(LayerConfigurationParameters.KernelHeight.class) final int kernelHeight,
+      @Parameter(LayerConfigurationParameters.KernelWidth.class) final int kernelWidth,
+      @Parameter(LayerConfigurationParameters.RandomSeed.class) final long randomSeed,
+      @Parameter(LayerConfigurationParameters.InitialWeight.class) final float initWeight,
+      @Parameter(LayerConfigurationParameters.InitialBias.class) final float initBias) {
+    this.matrixFactory = matrixFactory;
+    this.index = index;
+    this.inputShape = shapeFromString(inputShape);
+    this.paddingHeight = paddingHeight;
+    this.paddingWidth = paddingWidth;
+    this.strideHeight = strideHeight;
+    this.strideWidth = strideWidth;
+    this.kernelHeight = kernelHeight;
+    this.kernelWidth = kernelWidth;
+    this.outputShape = computeOutputShape();
+    this.randomSeed = randomSeed;
+    this.initWeight = initWeight;
+    this.initBias = initBias;
+  }
+
+  /**
+   * @return the initial parameter of the layer.
+   */
+  public LayerParameter generateInitialParameter() {
+    final Matrix weight = matrixFactory.randn(kernelHeight, kernelWidth, randomSeed);
+    final Matrix bias = matrixFactory.create(outputShape[0], outputShape[1]).fill(initBias);
+
+    weight.muli(initWeight); // multiply by standard deviation.
+
+    return LayerParameter.newBuilder()
+        .setWeightParam(weight)
+        .setBiasParam(bias)
+        .build();
+  }
+
+  /**
+   * @return the index of the layer.
+   */
+  public int getIndex() {
+    return index;
+  }
+
+  /**
+   * This function computes output shape.
+   * input shape: row * col
+   * output shape: row' * col'
+   * row' = (row − kernelHeight + 2 * paddingHeight) / stride + 1
+   * col' = (col − kernelWidth + 2 * paddingWidth) / stride + 1
+   * @return shape of output
+   */
+  private int[] computeOutputShape() {
+    final int[] computedShape = new int[2];
+    computedShape[0] = (int) Math.ceil((double) (inputShape[0] - kernelHeight + 2 * paddingHeight) / strideHeight) + 1;
```

Comment:
In addition, please add the code that an exception will be thrown when the input shape is not 2D.